src/Symfony/Component/Translation/Bridge/Loco/Tests/LocoProviderTest.php

```diff
@@ -226,114 +233,134 @@ public function testCompleteWriteProcess()
 
         $provider = $this->createProvider((new MockHttpClient($responses))->withOptions([
             'base_uri' => 'https://localise.biz/api/',
-            'headers' => [
-                'Authorization' => 'Loco API_KEY',
-            ],
+            'headers' => ['Authorization' => 'Loco API_KEY'],
         ]), $this->getLoader(), $this->getLogger(), $this->getDefaultLocale(), 'localise.biz/api/');
+
         $provider->write($translatorBag);
     }
 
     /**
-     * @dataProvider getLocoResponsesForOneLocaleAndOneDomain
+     * @dataProvider getResponsesForOneLocaleAndOneDomain
      */
     public function testReadForOneLocaleAndOneDomain(string $locale, string $domain, string $responseContent, TranslatorBag $expectedTranslatorBag)
     {
-        $response = $this->createMock(ResponseInterface::class);
-        $response->expects($this->once())
-            ->method('getContent')
-            ->willReturn($responseContent);
-        $response->expects($this->exactly(2))
-            ->method('getStatusCode')
-            ->willReturn(200);
-
         $loader = $this->getLoader();
         $loader->expects($this->once())
             ->method('load')
-            ->willReturn($expectedTranslatorBag->getCatalogue($locale));
+            ->willReturn((new XliffFileLoader())->load($responseContent, $locale, $domain));
```

Comment:
I'm not happy with this, but I don't find the correct way to either test it, or mock it. cc @nicolas-grekas (same in testReadForManyLocalesAndManyDomains)